python/heterocl/schedule.py

```diff
@@ -292,66 +286,298 @@ def to(self, tensors, dst, src=None, axis=0,
+            if self.hold_source_stage is not None:
+                src = self.hold_source_stage
+
+        # handle more than one input tensors for data movement 
         rets = list()
         if not isinstance(tensors, list):
             tensors = [tensors]
+
+        # handles many to many
+        if len(tensors) > 1: 
+            if isinstance(dst, list):
+                assert len(dst) == 1
+        if isinstance(dst, list):
+            if len(dst) == 1:
+                assert len(tensors) == 1
+
+        # one-to-many or many-to-one
+        # handle more than one dest (multi-casting)
+        if isinstance(dst, list):
+            for d in dst:
+                self.to(tensors, d, src, axis, mode, depth, burst, burst_len, name)
+            return self
+        
         for tensor in tensors:
             try:
+                # move the output tensor of a stage
                 if isinstance(tensor, Stage):
                     target = tensor._op
+
                 # unpack tuple of src stage and tensor
+                # E.g. kernel.stage.B = (stage, B)
                 elif isinstance(tensor, tuple):
                     src, target = tensor
-                    # from hcl stage to tvm stage
+                    # from heterocl stage to tvm stage
                     src = self.__getitem__(src)
+
                 else: # target tensor
                     target = tensor.tensor
-            except (AttributeError, ValueError):
+
+            except (AttributeError, ValueError) as e:
                 target = tensor
 
+                # if the src is already tvm stage
+                if isinstance(target, tuple):
+                    _, target = target
+
             # convert hcl stage
-            try: dst = self[dst]
+            try: 
+                if isinstance(dst, tuple):
+                   dst, _ = dst 
+                dst = self.__getitem__(dst)
             except: pass
 
             move_to_device = False
             if src is None:
                 # move to device
-                if isinstance(dst, Device) or \
-                        isinstance(dst, DevMediaPair):
+                # Example: s.to(A, target.FPGA)
+                if isinstance(dst, Device) or isinstance(dst, DevMediaPair):
                     if axis == 0:
                         move_to_device = True
                     else: # inner-stage movement
                         assert isinstance(tensor, Stage)
-                        target = self[tensor]
-
-                else: # inter-stage
-                    src = self[tensor]
+                        target = self.__getitem__(tensor)
+                    # burst_len == -1 means burst is diabled 
+                    if burst == True:
+                        if burst_len < 0: burst_len = 0
+                    else:
+                        assert burst_len == -1, "The burst mode must be True " + \
+                            "before setting the burst length..."
```

Comment:
This is too complicated. We should have `burst_len` only. If `burst_len` is -1, then `burst` is false. Otherwise `burst` is True. We don't need two arguments to config this.